### PR TITLE
Features/#278 multiext output support

### PR DESCRIPTION
**Pull request by dakochik**
Issue is resolved. 'Completion' and 'resolve' actions are working correctly with 'multiext' function.

Animation

### CODE REVIEW

File: src/main/kotlin/com/jetbrains/snakecharm/lang/psi/types/SmkRuleLikeSectionArgsType.kt [hunk 1]
```diff
@@ -164,11 +161,15 @@ class SmkRuleLikeSectionArgsType(
...
+     * E.g. "foo.txt" - one element
+     * multiext("f.", "t", "d") - two elements ["f.t", "f.d"]
+     *
+     * Note, that it supports smart counting only for 'multiext' function
      */
-    private fun isContainsExpandFunc(args: Array<out PyExpression>) = args.firstOrNull {
-        it is PyCallExpression && it.callee?.name.equals(SnakemakeNames.SNAKEMAKE_METHOD_EXPAND)
-    } != null
+    private fun countProducedElements(exp: PyExpression) = if (exp is PyCallExpression
+        && exp.callee?.name.equals(SnakemakeNames.SNAKEMAKE_METHOD_MULTIEXT)
```
Comment (iromeo) [inline src/main/kotlin/com/jetbrains/snakecharm/lang/psi/types/SmkRuleLikeSectionArgsType.kt:171]: please use `PyCallExpressin.isCalleeText(..)`

File: src/main/kotlin/com/jetbrains/snakecharm/lang/psi/types/SmkRuleLikeSectionArgsType.kt [hunk 2]
```diff
@@ -164,11 +161,15 @@ class SmkRuleLikeSectionArgsType(
     }
 
     /**
-     * Checks if there are any elements which are refer to [SnakemakeNames.SNAKEMAKE_METHOD_EXPAND] function
+     * Counts a number of elements that [PyExpression] returns.
+     * E.g. "foo.txt" - one element
+     * multiext("f.", "t", "d") - two elements ["f.t", "f.d"]
```
Comment (iromeo) [inline src/main/kotlin/com/jetbrains/snakecharm/lang/psi/types/SmkRuleLikeSectionArgsType.kt:166]: Let's use user-friendly example:
```
* multiext("foo", ".txt", ".log") - two elements ["foo.txt", "foo.log"]
```

File: src/test/resources/features/resolve/section_args_resolve.feature [hunk 3]
```diff
@@ -321,4 +321,31 @@ Scenario Outline: Unresolved index if out of bounds
...
+            "file",
+            <line3>
+         <exec_section>:
+            <key>
+        """
+      When I put the caret at <signature>
+      Then reference in injection should resolve to "<result>" in "foo.smk"
+      Examples:
+        | data_section | line1                    | line3                    | exec_section | key           | signature | result                   |
+        | input        | multiext("f.", "1", "2") | multiext("f.", "2", "1") | shell        | "{input[1]}"  | 1]}       | multiext("f.", "1", "2") |
```
Comment (iromeo) [inline src/test/resources/features/resolve/section_args_resolve.feature]: Here is better to resolve not just into `multiext(..)` call, but into it's corresponding agument

**Review by iromeo — CHANGES_REQUESTED** [review]
Resolve feature behaviour could be enhanced a bit + use PyCallExpression API to check call name.

**Comment by dakochik** [comment]
'resolve' behaviour update

multiext

File: src/test/resources/features/resolve/section_args_resolve.feature (hunk 3 above)
Comment (dakochik) [reply src/test/resources/features/resolve/section_args_resolve.feature]: Done

File: src/main/kotlin/com/jetbrains/snakecharm/lang/psi/types/SmkRuleLikeSectionArgsType.kt (hunk 2 above)
Comment (dakochik) [reply src/main/kotlin/com/jetbrains/snakecharm/lang/psi/types/SmkRuleLikeSectionArgsType.kt]: Done

File: src/main/kotlin/com/jetbrains/snakecharm/lang/psi/types/SmkRuleLikeSectionArgsType.kt (hunk 1 above)
Comment (dakochik) [reply src/main/kotlin/com/jetbrains/snakecharm/lang/psi/types/SmkRuleLikeSectionArgsType.kt]: Done

**Pull request merged**